addons/l10n_es_edi_tbai/models/account_move.py

```diff
@@ -0,0 +1,225 @@
+]
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    # Stored fields
+    l10n_es_tbai_chain_index = fields.Integer(
+        string="TicketBAI chain index",
+        help="Invoice index in chain, set if and only if an in-chain XML was submitted and did not error",
+        copy=False, readonly=True,
+    )
+
+    # Stored XML Binaries
+    l10n_es_tbai_post_xml = fields.Binary(
+        attachment=True, readonly=True, copy=False,
+        string="Submission XML",
+        help="Submission XML sent to TicketBAI. Kept if accepted or no response (timeout), cleared otherwise.",
+    )
+    l10n_es_tbai_cancel_xml = fields.Binary(
+        attachment=True, readonly=True, copy=False,
+        string="Cancellation XML",
+        help="Cancellation XML sent to TicketBAI. Kept if accepted or no response (timeout), cleared otherwise.",
+    )
+
+    # Non-stored fields
+    l10n_es_tbai_is_required = fields.Boolean(
+        string="TicketBAI required",
+        help="Is the Basque EDI (TicketBAI) needed ?",
+        compute="_compute_l10n_es_tbai_is_required",
+    )
+
+    # Optional fields
+    l10n_es_tbai_refund_reason = fields.Selection(
+        selection=[
+            ('R1', "R1: Art. 80.1, 80.2, 80.6 and rights founded error"),
+            ('R2', "R2: Art. 80.3"),
+            ('R3', "R3: Art. 80.4"),
+            ('R4', "R4: Art. 80 - other"),
+            ('R5', "R5: Factura rectificativa en facturas simplificadas"),
+        ],
+        string="Invoice Refund Reason Code (TicketBai)",
+        help="BOE-A-1992-28740. Ley 37/1992, de 28 de diciembre, del Impuesto sobre el "
+        "Valor Añadido. Artículo 80. Modificación de la base imponible.",
+        copy=False,
+    )
+
+    # -------------------------------------------------------------------------
+    # API-DECORATED & EXTENDED METHODS
+    # -------------------------------------------------------------------------
+
+    @api.depends('move_type', 'company_id')
+    def _compute_l10n_es_tbai_is_required(self):
+        for move in self:
+            move.l10n_es_tbai_is_required = move.is_sale_document() \
+                and move.country_code == 'ES' \
+                and move.company_id.l10n_es_tbai_tax_agency
+
+    @api.depends('state', 'edi_document_ids.state')
+    def _compute_show_reset_to_draft_button(self):
+        # EXTENDS account_edi account.move
+        super()._compute_show_reset_to_draft_button()
+
+        for move in self:
+            if move.l10n_es_tbai_chain_index:
+                move.show_reset_to_draft_button = False
+
+    def button_draft(self):
+        # EXTENDS account account.move
+        for move in self:
+            if move.l10n_es_tbai_chain_index and not move.edi_state == 'cancelled':
+                # TODO this last condition (state is cancelled) was not there in V15 and still we could cancel
+                # I'm super confused because I don't see why the behavior would change, please review
```

Comment:
@odoostan Normally, there is no big change with v15.  The question should be whether we allow a cancelled invoice to be reused or not.  Ideally, we do not and we can remove this condition. (but I think you tested and it works when you want to reuse an invoice)

Reply:
This is not about re-using/re-posting, it's about resetting to draft right ?
This states the invoice should first be cancelled to be able reset to draft. But it's inconsistent with the method above that determines the "reset to draft" button to be invisible when the invoice is on the chain.
Furthermore, cancelled invoices stay on the chain and keep their chain index. Therefore re-posting an invoice that was previously on the chain should never be possible.

Reply:
Then just remove the condition and the comment and we are good.  (after a pull/rebase)

The "and not move.edi_state == 'cancelled'" comes from the idea that you could reuse the move after a cancel, because if you put a cancelled move into draft, you can reuse it.